packages/plugin-compiler/src/main/kotlin/io/realm/compiler/RealmModelSyntheticPropertiesGeneration.kt

```diff
@@ -199,6 +205,14 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
         val fields: MutableMap<String, Pair<String, IrProperty>> =
             SchemaCollector.properties.getOrDefault(irClass, mutableMapOf())
 
+        val primaryKeyFields = fields.filter { it.value.second.backingField!!.hasAnnotation( PRIMARY_KEY_ANNOTATION) }
+
+        val primaryKey = when(primaryKeyFields.size)  {
+            0 -> ""
```

Comment:
This should be nullable. similarly `Table#primaryKey` 
```suggestion
            0 -> null
```
```Kotlin
val primaryKey: String? = null,
```